AdvancementAPI.java

```diff
@@ -71,66 +95,110 @@ public String getTitle() {
         return title;
     }
 
-    public AdvancementAPI withTitle(String title) {
+    public MessengerAdvancement withTitle(String title) {
         this.title = title;
         return this;
     }
 
+    public void setTitle(String title) {
+        this.title = title;
+    }
+
     public String getParent() {
         return parent;
     }
 
-    public AdvancementAPI withParent(String parent) {
+    public MessengerAdvancement withParent(String parent) {
         this.parent = parent;
         return this;
     }
 
+    public void setParent(String parent) {
+        this.parent = parent;
+    }
+
     public String getTrigger() {
         return trigger;
     }
 
-    public AdvancementAPI withTrigger(String trigger) {
+    public MessengerAdvancement withTrigger(String trigger) {
         this.trigger = trigger;
         return this;
     }
 
+    public void setTrigger(String trigger) {
+        this.trigger = trigger;
+    }
+
     public List<ItemStack> getItems() {
         return items;
     }
 
-    public AdvancementAPI withItem(ItemStack is) {
-        items.add(is);
+    public MessengerAdvancement withItem(ItemStack itemStack) {
+        items.add(itemStack);
         return this;
     }
 
+    public void addItem(ItemStack itemStack) {
+        items.add(itemStack);
+    }
+
     public FrameType getFrame() {
         return frame;
     }
 
-    public AdvancementAPI withFrame(FrameType frame) {
+    public MessengerAdvancement withFrame(FrameType frame) {
         this.frame = frame;
         return this;
     }
 
+    public MessengerAdvancement withFrame(String frame) {
+        if (frame.equalsIgnoreCase("random")) {
+            FrameType[] frames = FrameType.values();
+            setFrame(frames[(int)(Math.random()*(frames.length-1))]);
```

Comment:
move `frames[(int)(Math.random()*(frames.length-1))]` to FrameType. Then you're able to do FrameType.random()